include/openssl/core_dispatch.h

```diff
@@ -212,6 +212,21 @@ OSSL_CORE_MAKE_FUNC(int, provider_up_ref,
 OSSL_CORE_MAKE_FUNC(int, provider_free,
                     (const OSSL_CORE_HANDLE *prov, int deactivate))
 
+/* Functions to access the OBJ database */
+
+#define OSSL_FUNC_OBJ_ADD_SIGID                112
+#define OSSL_FUNC_OBJ_CREATE                   113
+#define OSSL_FUNC_OBJ_TXT2NID                  114
+#define OSSL_FUNC_OBJ_SN2NID                   115
+#define OSSL_FUNC_OBJ_LN2NID                   116
+
+OSSL_CORE_MAKE_FUNC(int, OBJ_add_sigid, (int signid, int dig_id, int pkey_id))
+OSSL_CORE_MAKE_FUNC(int, OBJ_create, (const char *oid, const char *sn,
+                                      const char *ln))
+OSSL_CORE_MAKE_FUNC(int, OBJ_txt2nid, (const char *s))
+OSSL_CORE_MAKE_FUNC(int, OBJ_sn2nid, (const char *s))
+OSSL_CORE_MAKE_FUNC(int, OBJ_ln2nid, (const char *s))
```

Comment:
Could these be reduced to:

```c
int OBJ_add_sig_by_name(const char *sign_name, const char *digest_name, const char *pkey_name);
int OBJ_create(const char *oid, const char *sn, const char *ln);
```
The first looks up the names and then calls OBJ_add_sigid() with them.  The second creates the object using names.  Both returns0/1 for fail/success.

The result would be no NIDs on the provider side.

Reply:
I strongly prefer this reduction.

Reply:
+1. Would it make sense to add a OPENSSL_CORE_CTX parameter to be able to do it per application libctx when/if the calling libcrypto has per-context object support?

Reply:
Actually yes, that would make sense.  We have that already for the ERR upcalls, so that would even be consistent :wink:

Reply:
I've reduced the API as suggested and added a `OSSL_CORE_HANDLE` parameter